components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -285,6 +285,19 @@ class BrowserToolbar @JvmOverloads constructor(
         edit.addEditAction(action)
     }
 
+    /**
+     * Removes an action that was displayed on the right of the URL in edit mode.
+     */
+    override fun removeEditAction(action: Toolbar.Action) {
+        edit.removeEditAction(action)
+    }
+
+    /**
+     * Checks if an action was displayed on the right of the URL in edit mode.
+     */
+    override fun containsEditAction(action: Toolbar.Action): Boolean =
+        edit.containsEditAction(action)
+
```

Comment:
I think the Fenix issue could also be modeled by using the action's `visible()` function to decide whether it should show up and use `invalidateActions()`?
Afaik the removal for page/browser actions was primarily added for addons where we permanently remove them - not switch between them.
What do you think?

Reply:
@pocmo you're right, didn't check yet for any visual glitches but it feels like the `invalidateActions` is the way to go, therefore I don't think there are any AC changes needed, thanks!